src/Features/Core/Portable/GenerateFromMembers/AbstractGenerateFromMembersService.cs

```diff
@@ -129,7 +129,36 @@ private static bool IsViableProperty(IPropertySymbol property)
 
         private static readonly char[] s_underscore = { '_' };
 
-        protected IMethodSymbol GetDelegatedConstructor(
+        /// <summary>
+        /// Try to find a constructor in <paramref name="containingType"/> whose parameters is the subset of <paramref name="parameters"/> by comparing name.
+        /// If multiple constructors meet the condition, the one with more parameters will be returned.
+        /// It will not consider those constructors as potential candidates if:
+        /// 1. Constructor with empty parameter list.
+        /// 2. Constructor's parameter list contains 'ref' or 'params'
+        /// </summary>
+        protected IMethodSymbol GetDelegatedConstructorBasedOnParameterNames(
+            INamedTypeSymbol containingType,
+            ImmutableArray<IParameterSymbol> parameters)
+        {
+            var parameterNames = parameters.SelectAsArray(p => p.Name);
+            return containingType.InstanceConstructors
+                .Where(constructor => IsParametersContainedInConstructor(constructor, parameterNames))
+                .OrderByDescending(constructor => constructor.Parameters.Length)
+                .FirstOrDefault();
+        }
+
+        private bool IsParametersContainedInConstructor(
```

Comment:
note: 'AreParametersContained...' would sound better as 'is' is singular but 'parameters' is plural.